### PR TITLE
[KYUUBI #304] Add Kyuubi Ctl arguments parser

### PR DESCRIPTION
**Pull request by turboFei**
turboFei [Closes #465](https://github.com/yaooqinn/kyuubi/pull/465)    Test Plan [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=yaooqinn&utm_campaign=badge_info)

### _Why are the changes needed?_
```
bin/kyuubi-service <create|get|delete|list>  <server|engine> --zkAddress ... --namespace ... --user ... --host ... --port ... --version

Operations:
    - create  -  expose a service to a namespace, this case is rare but sometimes we may want one server to be reached in 2 or more namespaces by different user groups
    - get - get the service node info
    - delete - delete the specified serviceNode
    - list - list all the service nodes for a particular domain
Role:
    - server default
    - engine 

Args:

   --zkAddress  - one of the zk ensemble address, using kyuubi-defaults/conf if absent
   --namespace  -  the namespace, using kyuubi-defaults/conf if absent
   --user - 
   --host
   --port
   --version
```

### _How was this patch tested?_
UT

### CODE REVIEW

File: kyuubi-assembly/pom.xml [hunk 1]
```diff
@@ -57,6 +57,12 @@
             <version>${project.version}</version>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.kyuubi</groupId>
```
Comment (yaooqinn) [inline kyuubi-assembly/pom.xml:61]: also add this to codecov module

Comment (yaooqinn) [reply kyuubi-assembly/pom.xml:61]: and we need to run `build/dependency.sh --replace` to update the list

File: kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+  }
+
+  private def printUsageAndExit(exitCode: Int, unknownParam: Any = null): Unit = {
+    if (unknownParam != null) {
+      info("Unknown/unsupported param " + unknownParam)
+    }
+    val command = sys.env.getOrElse("_KYUUBI_CMD_USAGE",
+      """Usage: kyuubi-ctl <create|get|delete|list>  <server|engine> --zkAddress ...
+        |--namespace ... --user ... --host ... --port ... --version""".stripMargin)
+    info(command)
```
Comment (yaooqinn) [inline kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala:115]: Seems better to use `println`?

File: kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+      case CREATE =>
+        action = KyuubiCtlAction.CREATE
+      case GET =>
+        action = KyuubiCtlAction.GET
+      case DELETE =>
+        action = KyuubiCtlAction.DELETE
+      case LIST =>
+        action = KyuubiCtlAction.LIST
+      case HELP =>
+        printUsageAndExit(0)
```
Comment (yaooqinn) [inline kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala:159]: maybe we can add a TODO here to make HELP action-specic too

Comment (turboFei) [reply kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala:159]: yes, it is better to treat HELP as an action.

File: kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+      info("Unknown/unsupported param " + unknownParam)
+    }
+    val command = sys.env.getOrElse("_KYUUBI_CMD_USAGE",
+      """Usage: kyuubi-ctl <create|get|delete|list>  <server|engine> --zkAddress ...
+        |--namespace ... --user ... --host ... --port ... --version""".stripMargin)
+    info(command)
+
+    info(
+      s"""
+         |Operations:
```
Comment (yaooqinn) [inline kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala:119]: Command

Comment (turboFei) [reply kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala:119]: Ok, I will refactor usage description.

File: kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+         |  - create                    expose a service to a namespace, this case is rare but
```
Comment (yaooqinn) [inline kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala:120]: Expose a service to a namespace on the zookeeper cluster of zkAddress manually

File: kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+         |                              namespaces by different user groups
+         |  - get                       get the service node info
+         |  - delete                    delete the specified serviceNode
+         |  - list                      list all the service nodes for a particular domain
+         |
+         |Role:
+         |  - server                    default
+         |  - engine
+         |
+         |Args:
```
Comment (yaooqinn) [inline kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala:131]: Arguements

File: kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+      s"""
+         |Operations:
+         |  - create                    expose a service to a namespace, this case is rare but
+         |                              sometimes we may want one server to be reached in 2 or more
+         |                              namespaces by different user groups
+         |  - get                       get the service node info
+         |  - delete                    delete the specified serviceNode
+         |  - list                      list all the service nodes for a particular domain
+         |
+         |Role:
```
Comment (yaooqinn) [inline kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala:127]: Services

File: kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+         |  - server                    default
+         |  - engine
+         |
+         |Args:
+         |  --zkAddress                 one of the zk ensemble address, using kyuubi-defaults/conf if absent
+         |  --namespace                 the namespace, using kyuubi-defaults/conf if absent
+         |  --user
+         |  --host
```
Comment (yaooqinn) [inline kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala:135]: hostname or IP address of a service

File: kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala [hunk 9]
```diff
@@ -0,0 +1,236 @@
...
+         |  --port
```
Comment (yaooqinn) [inline kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala:136]: listening port of a service

File: kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala [hunk 10]
```diff
@@ -0,0 +1,236 @@
...
+         |  --version
```
Comment (yaooqinn) [inline kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala:137]: --version      using the compiled KYUUBI_VERSION default, change it if the active service is running in another

File: pom.xml [hunk 11]
```diff
@@ -33,6 +33,7 @@
         <module>kyuubi-ha</module>
         <module>kyuubi-main</module>
         <module>kyuubi-metrics</module>
+        <module>kyuubi-ctl</module>
```
Comment (yaooqinn) [inline pom.xml:36]: nit: alphabet ordering

File: kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala (hunk 2 above)
Comment (turboFei) [reply kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala:119]: I think it does not matter.

Normally, the info level is INFO,CONSOLE, user can see this in console.
Even the info level is ERROR, user can see below KyuubiCtlException message.

BTW, now I test it based on the logAppender.
If I found a solution to transfer printStream into it gracefully, I will use printStream to replace it.

Comment (yaooqinn) [reply kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala:119]: make sense

**Comment by turboFei** [comment]
@yaooqinn 
Hi, I have try my best to increase code coverage.

**Comment by yaooqinn** [comment]
thanks, merged to master for v1.2.0, and wait for your followups

**Pull request closed without merging**